tracking_plan/yaml_event_property.py

```
@@ -31,8 +35,12 @@ def to_json(self):
         if self.allow_null:
             p_types.append('null')
 
-        return {
+        output = {
             'description': self.description,
             'type': p_types,
             'id': f"/properties/properties/properties/{self.name}"
-        }
\ No newline at end of file
+        }
+        if self.pattern:
```

Should we only allow the pattern when the `type` is String? From looking on the Protocols UI, they only allow regex w/ String type.

---

Good call @djfarrelly. What do you think we should do if they type is not `string` but a pattern is present? Should we raise an error or just not supply the pattern?

---

Yeah @michael-erasmus, I think we should raise and error. I think that could be useful as a way to implement validations too:

- If a required yaml key is missing (i.e. `type`) or an invalid value is supplied (a pattern for a non-string), we throw a custom `ValidationError` type
- If an error happens during an `update` - we fail and log the error and event name, property, etc.
- If an error happens during a new `validate` command, we catch the error and log it to the output so one could see all validation errors.

How does that sound?

---

@michael-erasmus, as a follow up, I think if we implement something like that, we can also run it on Travis for branch builds and fail if the validate fails